build.gradle

```diff
@@ -181,7 +181,10 @@ project(':ambry-api') {
         compile "io.dropwizard.metrics:metrics-core:$metricsVersion"
         compile "com.fasterxml.jackson.core:jackson-core:$jacksonVersion"
         compile "com.fasterxml.jackson.core:jackson-annotations:$jacksonVersion"
-        compile "io.netty:netty-buffer:$nettyVersion"
+        compile "io.netty:netty-buffer:$nettyVersion",
+        compile ("org.apache.helix:helix-core:$helixVersion") {
```

Comment:
it would be nice to keep helix out of ambry-api's dependencies if possible since it is used by ambry-client and helix's transitive dependencies have caused conflict in the past. (long term we can solve this problem by isolating the classes/interfaces that are actually needed by the client into a separate module and keep this as an internal-api module).

How hard do you think it would be to change `VcrClusterSpectator` to not implement helix interfaces and instead deal with ambry defined listener interfaces? It may be challenging since we probably have code that deals with raw `InstanceConfig` and `LiveInstance` objects. Ideally, we'd have ambry classes to represent vcr nodes like we do in the server clustermap so replication would not be coupled with helix concepts.

Reply:
I also have the plan to do change like you said. A separate PR for this change would be clear for reviewers. 
Moving out helix dependencies from ambry-api in this PR.

Reply:
Sure, sounds like a plan.  maybe we could put all these interfaces in the ambry-commons module for now. This will be like it was before where the module was ambry-commons and the java package was com.github.ambry.clustermap.